x/paramfilter/test/gov_params_test.go

```diff
@@ -0,0 +1,604 @@
+// modifiable in the specs params.md file cannot be modified via
+// governance.
+func (suite *GovParamsTestSuite) TestUnmodifiableParams() {
+	assert := suite.Assert()
+
+	testCases := []struct {
+		name         string
+		proposal     *proposal.ParameterChangeProposal
+		expectErr    bool
+		postProposal func()
+	}{
+		{
+			"bank.SendEnabled",
+			testProposal(proposal.ParamChange{
+				Subspace: banktypes.ModuleName,
+				Key:      string(banktypes.KeySendEnabled),
+				Value:    `[{"denom": "test", "enabled": false}]`,
+			}),
+			true,
+			func() {
+				got := suite.app.BankKeeper.GetParams(suite.ctx).SendEnabled
+				proposed := []*banktypes.SendEnabled{banktypes.NewSendEnabled("test", false)}
+				assert.NotEqual(proposed, got)
+			},
+		},
+		{
+			"consensus.block.TimeIotaMs",
+			testProposal(proposal.ParamChange{
+				Subspace: baseapp.Paramspace,
+				Key:      string(baseapp.ParamStoreKeyBlockParams),
+				Value:    `{"max_bytes": "1", "max_gas": "1", "time_iota_ms": "1"}`,
+			}),
+			// TimeIotaMs is not blocked by the param filter but it isn't
+			// exposed from CometBFT to the application so it can not be
+			// modified by governance. Since it isn't blocked by the param
+			// filter, we expect no error if a governance proposal includes it.
+			// We do expect the value to not change.
+			false,
+			func() {
+				got := suite.app.BaseApp.GetConsensusParams(suite.ctx).Block
+				proposed := tmproto.BlockParams{
+					MaxBytes:   1,
+					MaxGas:     1,
+					TimeIotaMs: 1,
+				}
+				assert.NotEqual(proposed, got)
+			},
+		},
+		{
+			"consensus.validator.PubKeyTypes",
+			testProposal(proposal.ParamChange{
+				Subspace: baseapp.Paramspace,
+				Key:      string(baseapp.ParamStoreKeyValidatorParams),
+				Value:    `{"pub_key_types": ["secp256k1"]}`,
+			}),
+			true,
+			func() {
+				got := *suite.app.BaseApp.GetConsensusParams(suite.ctx).Validator
+				proposed := tmproto.ValidatorParams{
+					PubKeyTypes: []string{"secp256k1"},
+				}
+				assert.NotEqual(proposed, got)
+			},
+		},
+		{
+			"staking.BondDenom",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyBondDenom),
+				Value:    `"test"`,
+			}),
+			true,
+			func() {
+				got := suite.app.StakingKeeper.GetParams(suite.ctx).BondDenom
+				proposed := "test"
+				assert.NotEqual(proposed, got)
+			},
+		},
+		{
+			"staking.UnbondingTime",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyUnbondingTime),
+				Value:    `"1"`,
+			}),
+			true,
+			func() {
+				got := suite.app.StakingKeeper.GetParams(suite.ctx).UnbondingTime
+				proposed := time.Duration(1)
+				assert.NotEqual(proposed, got)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			err := suite.govHandler(suite.ctx, tc.proposal)
+			if tc.expectErr {
+				suite.Require().Error(err)
```

Comment:
If errors could arise from causes other than the non-modifiability of parameters, it would be more appropriate to ascertain that the specific errors encountered are indeed due to the parameters' immutability

Reply:
oh I see, you're proposing asserting a specific error instead of just that an error was returned?

Reply:
Fixed in 46f453c85f8a69b9bd1c90fcdbad28f363484695